### PR TITLE
Moving changes from Development to Main

### PR DESCRIPTION
**Pull request by d-hmlton**
Implemented the coding task as described in the assignment brief ("Additional Information for Completion of Assessment")

### CODE REVIEW

**Comment by d-hmlton** [comment]
No conflicts with main, so merging performed automatically

File: CMP1903_A1_2324/Testing.cs [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
+            Debug.Assert(tesGame.Sum == -1);
+
+            //Ensuring the default value of the Die object's "DieVal" property is -1
+            Debug.Assert(tesDie.DieVal == -1);
+            
+            //Rolling die in our Game test object
+            tesGame.RollDie();
+
+            //Verifying the individual die values of the Game object
+            for (int i = 1; i < 4; i++) {
```
Comment (llama2474) [inline CMP1903_A1_2324/Testing.cs:45]: You dont need to test each individual dice, you only need to test 1 dice on its own and then one sum from the game.

File: CMP1903_A1_2324/Testing.cs [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
+namespace CMP1903_A1_2324 {
+    /// <summary>
+    /// Uses Debug.Assert() to perform a number of tests on the Game and Die classes.
+    /// </summary>
+    /// <remarks>
+    /// The first of these tests ensures the properties of both are correctly set to their default value, -1.
+    /// The result of the Game object's methods is then compared to their expected outputs. This means the dice rolls are
+    ///  checked to see if they're within 1-6, the sum to see if it's within 3-18, and the "Sum" property is validated by
+    ///  having the Testing class perform it's own sum calculation.
+    /// Similarly, the Die class is also checked to ensure the result is within 1-6, and the "DieVal" property is validated
```
Comment (Cait43) [inline CMP1903_A1_2324/Testing.cs:17]: think you might have repeated yourself a little bit here

File: CMP1903_A1_2324/Game.cs [hunk 3]
```diff
@@ -4,18 +4,45 @@
...
+            int secondRoll = dieSecond.Roll(); Console.WriteLine("Die 2: " + secondRoll);
+            int thirdRoll = dieThird.Roll(); Console.WriteLine("Die 3: " + thirdRoll);
+            _sum = firstRoll + secondRoll + thirdRoll;
+            Console.WriteLine("SUM = " + _sum);
+        }
+
+        public int GetRolls(int die) {
+            if (die < 1 || die > 3) {
+                throw new ArgumentOutOfRangeException(die + " is not in the accepted range.");
+            }
```
Comment (Cait43) [inline CMP1903_A1_2324/Game.cs:38]: could use some comments to explain this part of the code

**Review by Cait43 — COMMENTED** [review]
You have met all the requirements of the brief and included all necessary aspects of the code.

The layout of the code was difficult to follow at times, with multiple objects all being listed in one line instead of underneath. Additionally, some parts of the code could use commenting for a more direct explanation than the summary at the top. Overall, I was able to understand the code, the layout was just something I wasn't used to and took a bit longer to understand fully. 

Errors appear to be sufficiently handled using the Debug.Assert() command 

Good use of correct naming conventions throughout the code; the proper formatting appears to be followed with naming conventions, class orders  and group elements all fitting within the specification.

**Pull request merged**